includes/discovery/discovery-protocols.inc.php

```diff
@@ -81,7 +81,47 @@
     echo PHP_EOL;
 }//end if
 
-if (($device['os'] == 'pbn' || $device['os'] == 'bdcom') && Config::get('autodiscovery.xdp') === true) {
+if (($device['os'] == 'routeros') && Config::get('autodiscovery.xdp') === true) {
+    echo ' LLDP-MIB: ';
+    $lldp_array  = snmpwalk_group($device, 'lldpRemEntry', 'LLDP-MIB', 3);
+    if (!empty($lldp_array)) {
+        $lldp_ports = snmpwalk_group($device, 'mtxrInterfaceStatsName', 'MIKROTIK-MIB');
+        $lldp_ports_num = snmpwalk_group($device, 'mtxrNeighborInterfaceID', 'MIKROTIK-MIB');
+    }
```

Comment:
May as well move the `foreach ($lldp_array)` into this if check, no point in even attempting to loop through the empty array.